ledger/index.ts

```diff
@@ -336,65 +322,196 @@ export async function* signLedgerMixedBtcTransaction(
 }
 
 /**
- * This function is used to sign an incoming Native Segwit Psbt with the ledger
+ * This function is used to sign an incoming Native Segwit / Taproot PSBT with the ledger
  * @param transport - the transport object with connected ledger device
  * @param network - the network type (Mainnet or Testnet)
  * @param addressIndex - the index of the account address to sign with
- * @param indexesToSign - the indexes of the inputs to sign
- * @param base64Psbt - the incoming transaction in base64 format
- * @returns the signed raw transaction in hex format
+ * @param inputsToSign - the array of inputs to sign with address and indexes
+ * @param psbtBase64 - the incoming transaction in base64 format
+ * @param finalize - boolean to finalize the transaction
+ * @returns the signed PSBT in string (base64) format
  * */
-export async function signIncomingSingleSigNativeSegwitTransactionRequest(
+export async function signIncomingSingleSigPSBT(
   transport: Transport,
   network: NetworkType,
   addressIndex: number,
-  indexesToSign: number[],
-  base64Psbt: string
+  inputsToSign: InputToSign[],
+  psbtBase64: string,
+  finalize = false,
 ): Promise<string> {
+  if (!psbtBase64?.length) {
+    throw new Error('Invalid transaction');
+  }
+
   const coinType = getCoinType(network);
   const app = new AppClient(transport);
 
+  const types = inputsToSign.map((input: InputToSign) => {
```

Comment:
we are already mapping the inputs and checking the type when we update the derivation so this here is redundant we can just set the flags over there